modules/gdnative/include/gdnative/aabb.h

```diff
@@ -35,9 +35,10 @@
 extern "C" {
 #endif
 
+#include "core/math/math_defs.h"
```

Comment:
These header files under `include` are meant to be used by GDNative libraries, so they should be self-contained and thus cannot depend on `core`. Not sure how to solve this but maybe we could have a `math_defs.h` in GDNative itself to solve what `real_t` is without doing it in every header.

Reply:
Actually, there is already `typedef float godot_real;` in `modules/gdnative/include/gdnative/gdnative.h`. This should probably be changed to this:

```cpp
#ifdef REAL_T_IS_DOUBLE
typedef float godot_real;
#else
typedef float godot_real;
#endif
```

Then we need to move up `#include <gdnative/gdnative.h>` in these headers, and then `sizeof(real_t)` can be `sizeof(godot_real)`.

Reply:
This was removed. `godot_real` is renamed to `godot_float` and it's always a `double` (since it's meant to be used in Variant which is always 64 bits). We may define a `godot_real_t` to use here instead.

And yes, you cannot include `gdnative.h` before the definition of the structs, that's why I suggested a new header.

> This assumes that people including this would define `REAL_T_IS_DOUBLE` in their GDNative builds if they use doubles.

Yes, people will have to that anyway.